--- xf86-video-xgi-1.5.0/src/xgi_driver.c.orig	2007-08-28 00:56:34.000000000 +0200
+++ xf86-video-xgi-1.5.0/src/xgi_driver.c	2008-12-27 10:49:05.489307405 +0100
@@ -2504,7 +2504,7 @@
 
     FbDevExist = FALSE;
     if (pXGI->Chipset != PCI_CHIP_XGIXG20) {
-        if ((fd = open("/dev/fb", 'r')) != -1) {
+        if ((fd = open("/dev/fb", O_RDONLY)) != -1) {
             PDEBUG(ErrorF("--- open /dev/fb....   \n"));
             ioctl(fd, FBIOGET_FSCREENINFO, &fix);
             if (fix.accel == FB_ACCEL_XGI_GLAMOUR) {
